pkgs/tools/archivers/undmg/setup-hook.sh

```diff
@@ -1,5 +1,25 @@
 unpackCmdHooks+=(_tryUnpackDmg)
 _tryUnpackDmg() {
     if ! [[ "$curSrc" =~ \.dmg$ ]]; then return 1; fi
-    undmg "$curSrc"
+    dmgFs="$(/usr/bin/hdiutil imageinfo "$curSrc" | sed -n '/Name: disk image/s/.*(Apple_\([^ ]*FS\).*)/\1/p')"
+    case "$dmgFs" in
+        "HFS") undmg "$curSrc";;
+        "APFS")
+            mnt=$(mktemp -d -t ci-XXXXXXXXXX)
+            function finish {
+                echo "Unmounting $mnt and detaching $curSrc"
+                /usr/bin/hdiutil detach $mnt -force
+                rm -rf $mnt
+            }
+            trap finish EXIT
```

Comment:
the function name should be more specific than finish

Reply:
Will do.

Reply:
Done, do you agree with the naming choice?

Reply:
With the latest changes to this PR this is no longer relevant.